File: 1143-longest-common-subsequence/1143-longest-common-subsequence.js
/**
 * @param {string} text1
 * @param {string} text2
 * @return {number}
 */
var longestCommonSubsequence = function(text1, text2) {
    
    let dp = new Array(text1.length);
    for(let i = 0; i <= text1.length; i++){
        dp[i] = new Array(text2.length+1);
    }
    
    for(let j=0;j<dp.length;j++){
        dp[j][0]=0;
    }
    
    for(let k=0;k<dp[0].length;k++){
        dp[0][k]=0;
    }
    
    for (let row = 1; row < dp.length; row++) {
        for (let col = 1; col < dp[0].length; col++) {
            if(text1[row-1] === text2[col-1]){
                // console.log(row,col)
                dp[row][col] = dp[row-1][col-1] + 1;
            } else {
                dp[row][col] = Math.max( dp[row-1][col], dp[row][col-1] );
            }
        }
    };
    
    // console.log(dp);
    return dp.at(-1).at(-1);
};

File: 1143-longest-common-subsequence/1143-longest-common-subsequence.py
class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        text1 = '#' + text1
        text2 = '*' + text2
        m,n = len(text1), len(text2)
        
        dp = [[0 for _ in range(n)] for _ in range(m)]
        
        for i in range(1,m):
            for j in range(1,n):
                if text1[i] == text2[j]:
                    dp[i][j] = 1 + dp[i-1][j-1]
                else:
                    dp[i][j] = max(dp[i-1][j], dp[i][j-1])
        print(dp)
        return dp[-1][-1]
                                                    